text/0140-record-feature-flags-on-error.md

```diff
@@ -0,0 +1,47 @@
+- Start Date: 2024-09-04
+- RFC Type: feature
+- RFC PR: https://github.com/getsentry/rfcs/pull/140
+- RFC Status: draft
+
+# Summary
+
+As feature flags are evaluated in a customer's application we will collect those flags and their results, store them in-memory, and push the values to Sentry when an error occurs. The flags can be used to further understand what went wrong during the session.
+
+# Motivation
+
+To enable users to debug application errors with a complete picture of their application's state.
+
+# Options Considered
+
+We will collect flag evaluations and hold them in-memory. On error event the flags will be placed into the `contexts` object on the `event` and sent to Sentry.
+
+## Transport
+
+The flags will be represented by this data structure during transport. The `flag` key is the name of the flag and the `result` key is the evaluation result returned by the customer's application.
+
+```json
+{
+  "contexts": {
+    "flags": {
+      "values": [
+        { "flag": "abc", "result": true },
+        { "flag": "def", "result": false }
+      ]
+    }
+  }
+}
+```
+
+## Public Interface
+
+The SDK will expose one new public method `set_flag`. The method accepts the arguments `flag` (of type string) and `result` (of type boolean). Similar to `set_tag` or `set_user`, the `set_flag` method stores a flag, result pair on the isolation scope. On error, the isolation scope's flags are serialized and appended to the event body as described in the previous section.
```

Comment:
What if the user uses a number or a string as feature flag?
[LaunchDarkly](https://docs.launchdarkly.com/sdk/concepts/flag-types) (and probably others) supports that